python_modules/dagster/dagster/core/execution.py

```diff
@@ -223,18 +205,31 @@ def success(self):
     @property
     def transformed_values(self):
         '''Return dictionary of transformed results, with keys being output names.
-        Returns None if execution result isn't a success.'''
+        Returns None if execution result isn't a success.
+
+        Reconstructs the pipeline context to materialize values. Results are memoized so
+        only the first call will reconstruct the context.
+        '''
         if self.success and self.transforms:
-            return {
-                result.step_output_data.output_name: result.step_output_data.get_value()
-                for result in self.transforms
-            }
+            if self._memoized_values is None:
```

Comment:
I actually think memoizing is probably not worth it and we should just rely on callers to cache if we want to. There are cases where we may be outputing large data sets to disk and this could be very expensive to load them back in, especially if they are unused.